server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSPerformSPOptionsCheckUtility.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2025 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.implicitDialects.cics.utility;
+
+        import org.antlr.v4.runtime.ParserRuleContext;
+
+        import org.eclipse.lsp.cobol.common.dialects.DialectProcessingContext;
+        import org.eclipse.lsp.cobol.common.error.ErrorSeverity;
+        import org.eclipse.lsp.cobol.common.error.SyntaxError;
+        import org.eclipse.lsp.cobol.implicitDialects.cics.CICSLexer;
+        import org.eclipse.lsp.cobol.implicitDialects.cics.CICSParser;
+
+        import java.util.HashMap;
+        import java.util.List;
+        import java.util.Map;
+
+        import static org.eclipse.lsp.cobol.implicitDialects.cics.CICSParser.*;
+
+/** Checks CICS Create System Command rules for required and invalid options */
+public class CICSPerformSPOptionsCheckUtility extends CICSOptionsCheckBaseUtility {
+
+    public static final int RULE_INDEX = RULE_cics_perform;
+    private static final Map<Integer, ErrorSeverity> DUPLICATE_CHECK_OPTIONS =
+            new HashMap<Integer, ErrorSeverity>() {
+                {
+                    put(CICSLexer.DELETSHIPPED, ErrorSeverity.WARNING);
+                    put(CICSLexer.DUMPCODE, ErrorSeverity.ERROR);
+                    put(CICSLexer.DUMP, ErrorSeverity.WARNING);
```

Comment:
Duplicate DUMP is error in `PERFORM JVMSERVER`. This keyword might need special handling.